wallet/src/components/IbcTransfer.tsx

```diff
@@ -0,0 +1,497 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { useEffect, useMemo, useState } from 'react';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import InputAdornment from '@mui/material/InputAdornment';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import { parseAsValue } from '@agoric/ui-components';
+import { withApplicationContext } from '../contexts/Application';
+import { ibcAssets } from '../util/ibc-assets';
+import { stringifyPurseValue } from '@agoric/ui-components';
+import ArrowDownward from '@mui/icons-material/ArrowDownward';
+import { Box } from '@mui/system';
+import { fromBech32 } from '@cosmjs/encoding';
+import { queryBankBalances } from '../util/queryBankBalances';
+import { assertIsDeliverTxSuccess } from '@cosmjs/stargate';
+import { CircularProgress, Link, Snackbar, Typography } from '@mui/material';
+import Petname from './Petname';
+import type { PurseInfo } from '../service/Offers';
+import type { KeplrUtils } from '../contexts/Provider';
+import type { Petname as PetnameType } from '@agoric/smart-wallet/src/types';
```

Comment:
since `Petname` is type that spans repos, consider letting it win this naming collision. `Petname.tsx` could be `PetnameSpan`.

Reply:
Done